src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposable.cs

```diff
@@ -87,16 +87,18 @@ where _disposeAnalysisHelper.IsDisposable(field.Type)
                 foreach (TTypeDeclarationSyntax classDecl in classDecls)
                 {
                     SemanticModel model = symbolContext.Compilation.GetSemanticModel(classDecl.SyntaxTree);
-                    IEnumerable<string> disposableFieldNames = classDecl.DescendantNodes(n => !(n is TTypeDeclarationSyntax) || ReferenceEquals(n, classDecl))
+                    List<string> disposableFieldNames = classDecl.DescendantNodes(n => n is not TTypeDeclarationSyntax || ReferenceEquals(n, classDecl))
                         .SelectMany(n => GetDisposableFieldCreations(n, model, disposableFieldsHashSet, symbolContext.CancellationToken))
                         .Where(field => !symbolContext.Options.IsConfiguredToSkipAnalysis(Rule, field.Type, namedType, symbolContext.Compilation))
-                        .Select(field => field.Name);
+                        .Select(field => field.Name)
+                        .ToList();
```

Comment:
❓ @mavasani  Should we be making this a symbol start/stop analyzer instead of trying to walk descendant nodes in a symbol analyzer?

Reply:
Definitely, seems like a good candidate to register a nested callback for field declaration nodes in the SymbolStart callback, which would collect all disposable field names, and the SymbolEnd callback can report the diagnostic.

Reply:
Done.